pycbc/events/stat.py

```diff
@@ -2107,12 +2133,29 @@ def lognoiserate(self, trigs):
 
         Returns
         ---------
-        lognoisel: numpy.array
+        lognoiserate: numpy.array
             Array of log noise rate density for each input trigger.
         """
+
+        # make sure every trig has a dq state
+
+        try:
+            # works in offline
+            ifo = trigs.ifo
+        except AttributeError:
+            ifo = trigs['ifo']
+            unq = numpy.unique(ifo)
+            assert len(unq) == 1
+            # Should be exactly one ifo provided
+            ifo = unq[0]
```

Comment:
is this equivalent to `ifo = ifo[0]` ? 
If so, it is probably simpler not to introduce a new variable .. ie `assert len(numpy.unique[ifo]) == 1` then `ifo = ifo[0]`